bee-pow/src/providers/miner.rs

```diff
@@ -107,19 +107,21 @@ impl Provider for Miner {
     type Builder = MinerBuilder;
     type Error = Error;
 
-    fn nonce(&self, bytes: &[u8], target_score: f64) -> Result<u64, Self::Error> {
+    fn nonce(&self, bytes: &[u8], target_score: f64, done: Option<Arc<AtomicBool>>) -> Result<u64, Self::Error> {
         let mut nonce = 0;
         let mut blake = VarBlake2b::new(32).unwrap();
         let mut pow_digest = TritBuf::<T1B1Buf>::new();
-
+        let done = match done {
+            Some(d) => d,
+            None => Arc::new(AtomicBool::new(false)),
+        };
```

Comment:
Can you refactor in an `unwrap_or_else` ?

Reply:
Yes, will do

Reply:
Should I do it like `let done = done.unwrap_or_else(|| Arc::new(AtomicBool::new(false)));`?

Reply:
Yes, the reason we prefer that over `unwrap_or(Arc::new(AtomicBool::new(false)))` is that it's lazily evaluated, meaning the content or the or_else is only executed in case of None.

Reply:
`unwrap_or` is fine when the value is a constant and not from a potentially expensive call

Reply:
Thanks for the explanation